subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/execution/PartialEvaluator.kt

```diff
@@ -32,10 +32,12 @@ enum class ProgramKind {
 }
 
 
-enum class ProgramTarget {
-    Project,
-    Settings,
-    Gradle
+enum class ProgramTarget(
+    val supportsPluginsBlock: Boolean
```

Comment:
❌ There is a single place where this value is actually used. It doesn't make sense to force it upon all clients of this data type. Move the computation to where is actually used (`ProgramParser`).